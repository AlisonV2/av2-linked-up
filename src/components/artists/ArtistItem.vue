<template>
  <div class="card artist-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title">{{ artist.name }}</h5>
      <p class="card-text">{{ artist.shop }}</p>
      <p class="card-text">{{ artist.city }} ({{ artist.zip }})</p>
    </div>
    <img
      :src="artist.thumbnail"
      class="card-img-top"
      :alt="artist.name"
    />
    <div class="card-body">
      <h5 class="card-title">{{ artist.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @type Component
 * @name ArtistItem
 * @vue-prop {object} artists
 */
export default {
  name: 'ArtistItem',
  props: ['artist'],
};
</script>

<style lang="scss" scoped>
.artist-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.artist-card:hover {
  .card-details {
    display: block;
  }
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
