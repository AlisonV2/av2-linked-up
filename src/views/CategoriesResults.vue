<template>
  <div class="container">
  <div class="row">
    <div
      class="col-8 col-md-6 col-lg-4 col-xl-3 artists-list"
      v-for="artist in artists"
      :key="artist.name"
    >
      <router-link :to="{ name: 'Profile', params: { id: artist.id } }">
        <ArtistItem :artist="artist" />
      </router-link>
    </div>
  </div>
  </div>
</template>

<script>
import ArtistItem from '@/components/artists/ArtistItem';

export default {
  name: 'CategoriesResults',
  components: {
    ArtistItem,
  },
  data() {
    return {
      artists: []
    };
  },
  created() {
      this.artists = this.$store.state.categories.artists;
  }
};
</script>
