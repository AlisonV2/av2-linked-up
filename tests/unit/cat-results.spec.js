import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import CategoriesResults from '@/views/CategoriesResults.vue';
import ArtistItem from '@/components/artists/ArtistItem';

const artists = [
  {
    id: '123',
    name: 'Dirty Jonz',
    shop: 'Le Dernier Rituel',
    city: 'BÃ¨gles',
    zip: '33130',
  },
];

const config = {
  data() {
    return { artists: artists };
  },
  global: {
    components: {
      ArtistItem: ArtistItem,
      'router-link': RouterLinkStub,
    },
    mocks: {
      $store: {
        state: {
          search: {
            artists: artists,
          },
        },
      },
    },
  },
};

/**
 * @module CategoriesResultsTest
 */
describe('CategoriesResults.vue', () => {
  it('Categories Snapshot', () => {
    const component = shallowMount(CategoriesResults, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Profile link should be rendered', () => {
    const component = shallowMount(CategoriesResults, config);
    expect(component.findComponent(RouterLinkStub).props().to).toEqual({
      name: 'Profile',
      params: { id: artists[0].id },
    });
  });

  it('Back to home link should be rendered', () => {
    const component = shallowMount(CategoriesResults, {
      data() {
        return { artists: [] };
      },
      global: {
        components: {
          ArtistItem: ArtistItem,
          'router-link': RouterLinkStub,
        },
        mocks: {
          $store: {
            state: {
              search: {
                artists: [],
              },
            },
          },
        },
      },
    });
    expect(component.findComponent(RouterLinkStub).props().to).toEqual({
      name: 'Home',
    });
  });
});
