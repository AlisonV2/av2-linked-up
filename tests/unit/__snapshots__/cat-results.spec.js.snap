// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesResults.vue Categories Snapshot 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    
    <div
      class="col-8 col-md-6 col-lg-4 col-xl-3 artists-list"
    >
      <router-link-stub
        to="[object Object]"
      />
    </div>
    
  </div>
</div>
`;
